### PR TITLE
Add brute force box fitting period search

### PR DESCRIPTION
**Pull request by mirca**
See https://notebooks.azure.com/mirca/libraries/box-period-search/html/Untitled.ipynb
cc @barentsen

### CODE REVIEW

**Comment by barentsen** [comment]
Wooowww that's such an elegant and easy-to-read implementation!

You probably know that the original BLS paper (https://www.aanda.org/articles/aa/pdf/2002/31/aa2422.pdf) presents algebra to speed up the computation, but as a result, BLS implementations are hard to read and somewhat tricky to implement correctly.  It's really cool to have this elegant implementation.  At minimum, it will be useful to test the correctness of faster (and more cryptic) BLS implementations.  And maybe we can even find a way to make this implementation nearly as quick anyway.

Awesome job!!

**Comment by mirca** [comment]
@barentsen Thanks!!! Yeah, *there must be a way* to speed things up... I'm thinking about improving this `box` function or relaxing the `options` which is passed to `scipy.minimize`

**Comment by mirca** [comment]
Food for thought:  L1-norm for the first graph, L2-norm for the second graph

screen shot 2018-01-20 at 8 15 20 pm
screen shot 2018-01-20 at 8 15 28 pm

Definitely need to experiment.

**Comment by mirca** [comment]
Example on Kepler-699b

L1-Norm
screen shot 2018-01-21 at 5 56 02 am

L2-Norm 
screen shot 2018-01-21 at 6 05 24 am

**Comment by barentsen** [comment]
Wow, why is the L2 norm so much better (?) In the second case?

**Comment by mirca** [comment]
@barentsen not sure, but I would suggest to use the l2-norm 👍

File: pyke/lightcurve.py [hunk 1]
```diff
@@ -1067,3 +1067,63 @@ def _pld(self, tpf_flux, polyorder=2):
...
+        pass
+
+    def search(self, lc, minper, maxper, nperiods=2000, prior=None):
+        """
+        Parameters
+        ----------
+        lc : LightCurve object
+            An object from KeplerLightCurve or LightCurve
+        minper : float
+            Minimum period to search for.
```
Comment (barentsen) [inline pyke/lightcurve.py]: You could say that the units must be the same as the units of `lc.time`.

File: pyke/lightcurve.py [hunk 2]
```diff
@@ -1067,3 +1067,63 @@ def _pld(self, tpf_flux, polyorder=2):
...
+        maxper : float
+            Maximum period to search for.
```
Comment (barentsen) [inline pyke/lightcurve.py]: Should we name these arguments `min_period` and `max_period` for clarity, or am I being too verbose?

File: pyke/lightcurve.py [hunk 3]
```diff
@@ -1067,3 +1067,63 @@ def _pld(self, tpf_flux, polyorder=2):
...
+        nperiods : int
+            Number of periods to search between `minper` and `maxper`.
+        prior : Prior object
+            Prior probability on the parameters of the box function.
```
Comment (barentsen) [inline pyke/lightcurve.py]: I'd say `oktopus.Prior` object instead of just `Prior` object.

File: pyke/lightcurve.py (hunk 2 above)
Comment (mirca) [reply pyke/lightcurve.py]: Yes, let's do it!

File: pyke/lightcurve.py (hunk 3 above)
Comment (mirca) [reply pyke/lightcurve.py]: Right!

File: pyke/lightcurve.py [hunk 4]
```diff
@@ -1067,3 +1067,63 @@ def _pld(self, tpf_flux, polyorder=2):
...
+class BoxLikePeriodSearch(object):
+    def __init__(self):
+        pass
+
+    def search(self, lc, minper, maxper, nperiods=2000, prior=None):
+        """
+        Parameters
+        ----------
+        lc : LightCurve object
+            An object from KeplerLightCurve or LightCurve
```
Comment (barentsen) [inline pyke/lightcurve.py]: Is it important that `lc` is flattened?  If so, we should say it in the docstring.

Comment (mirca) [reply pyke/lightcurve.py]: Yes, that's important

File: pyke/lightcurve.py (hunk 1 above)
Comment (mirca) [reply pyke/lightcurve.py]: 👍

**Comment by mirca** [comment]
Now  I realize there is no point in having `BoxLikePeriodSearch` as a class.
Should the function name be `box_period_search`?

**Comment by barentsen** [comment]
Yes, let's make it a function!  `box_period_search` is good. Could also consider naming it `brute_force_period_search` to distinguish it from the classic BLS algorithm.

**Comment by barentsen** [comment]
@mirca Oops, your latest commit added `pyke/Untitled.ipynb`.

**Pull request merged**